### PR TITLE
Backport MultiCraft version

### PR DESCRIPTION
**Pull request by MoNTE48**
On @sofar request.
1) Useless changes
2) Possible bugs
3) Texture Compression
4) Extra copyright
5) Removing dependencies with MT doors mod

### CODE REVIEW

**Comment by MoNTE48** [comment]
It could close https://github.com/minetest-mods/flowerpot/issues/3, if it will be merged

**Comment by sofar** [comment]
I'll need to test, noting unreasonable though, so it looks OK

File: init.lua
Comment (SmallJoker) [inline init.lua]: This will match all nodes registered within https://github.com/TalkLounge/plot_system 
Great job, but please make these exceptions a separate hard-coded list to not get false-positives.

Comment (MoNTE48) [reply init.lua]: I didn’t quite understand which of my filters might not work as expected. Clarify, please.

Comment (SmallJoker) [reply init.lua]: The issue is that string searching in node names is not reliable. You only want plants to have a flowerpot.
Nodes like `mymod:system_fan_controller` or `itemstemperature:foobar` would get a flower pot registration. That's definitely not intended.

Comment (MoNTE48) [reply init.lua]: You're right. I will come up with a more reliable solution. Thanks for the review.

Comment (MoNTE48) [reply init.lua]: @SmallJoker
**or string.sub(node, -5)== "_stem"**

Comment (SmallJoker) [reply init.lua]: That's already better, although adding a new group to the node definition (MTG change) would be the best solution.

I think for now it's okay to use `string.sub(node, -5)== "_stem"` even if there might still be rare false-positives.

Comment (MoNTE48) [reply init.lua]: Fixed

**Comment by MoNTE48** [comment]
Fixed found @SmallJoker problem and made several cleanings.

If everything is ok, then maybe after **4+ months** this simplest PR can be merged?

**Review by SmallJoker — APPROVED** [review]
Not perfect but good enough.

**Comment by SmallJoker** [comment]
Will merge in a few days unless there are objections from other devs.

**Comment by MoNTE48** [comment]
> Not perfect but good enough.

It's offensive. :(
If you want, I can check the _descriptions_ of items for the presence of "Stem" case-sensitive (bad idea, I know), or check only the "default" mod on "_stem" nodes.

**Comment by SmallJoker** [comment]
@MoNTE48 In this PR it's the best you could do without making it more hacky. The only proper solution would be to add groups to the stems in MTG but that's definitely out of scope. It's not meant to be offensive FYI.

File: LICENSE
Comment (sofar) [inline LICENSE:9]: please remove this, none of the contributions are significant enough to warrant an additional copyright claim.

File: init.lua
Comment (sofar) [reply init.lua]: I really don't like this wildcard method. You can trivially create e a secondary mod that adds registrations for wildcard nodes instead. I prefer much to not have a wildcard instead.

**Comment by sofar** [comment]
The described changes need to be in the git commit message. You are only putting some of the notices in the github thread, which is great, but doesn't absolve you from writing decent git commit messages.

I'm opposed to wildcard registrations and almost anything you're doing in this change because it feels like you're just offloading bad practices to this mod. The copyright change is frankly entirely out of place. There are far better ways as I wrote to integrate this mod into multicraft - a simple layer mod that depends on frame and multicraft mods adding more nodes would do the trick, or you could just depend on `frame` in code multicraft mods that add nodes given that this mod has very little dependencies to begin with.

:-1:

**Comment by MoNTE48** [comment]
Well, MultiCraft has its own version of everything.  In MultiCraft there is not a single mod that has not been changed.
If my simplification is _bad practice_, then I suggest closing this PR.

I will use this "_bad practice_" in my game and beyond. Thanks for giving some “_good practice_” to my big mess.

As for the **commit message**, English is not my native language, so I leave a good message to someone who will do the merge.

**Comment by sofar** [comment]
rejected.

**Pull request closed without merging**
